validphys2/src/validphys/coredata.py

```diff
@@ -9,6 +9,117 @@
 import pandas as pd
 
 
+def _get_or_empty(df, keys=[]) -> pd.DataFrame:
+    """
+    Extracts information from a dataframe as given by keys.
+    If any key is not found within the dataframe, returns an empty dataframe
+    with the right indices so that it can still be manipulated.
+
+    For instance: ``_get_or_empty(df, keys=["data", "DIS", "NMC"])`` will
+    return ``df["data"]["DIS"]["NMC"]``
+    or an empty dataframe if any of the 3 keys is not found.
+
+    Parameters
+    ----------
+        df: pd.DataFrame
+            dataframe from which extract the values
+        keys: list(str)
+            list of keys to iteratively extract values from the dataframe
+
+    Returns
+    -------
+        pd.DataFrame
+    """
+    """Return dataframe consuming all keys or an empty one if some key is
+    not found"""
+    if keys:
+        try:
+            return _get_or_empty(df[keys[0]], keys=keys[1:])
+        except KeyError:
+            return pd.DataFrame([], index=df.index)
+    return df
+
+
+@dataclasses.dataclass
+class _FancyDataframe:
+    """Holds a dataframe in a dataclass to which vp cuts can be applied"""
+
+    dataframe: pd.DataFrame
+
+    def __post_init__(self):
```

Comment:
Doesn't seem needed

Reply:
It looks I forgot that here at some point.

Did you look at the jupyter notebook? I would be interested on your opinion about how the data get saved and "understood" in terms of the dataframes. The methods, attributes and so on we can polish afterwards.

Reply:
Ah, no, it's needed because otherwise children classes won't call `__post_init__`. Don't know whether I did something wrong in the children classes, but that's the reason.

Reply:
The children classes should have the dataclass decorator as well.

Reply:
I guess the `@dataclass` is a decorator somehow `vp`-like: it reads you're code, and generate some other code, e.g. the `__init__` function, out of it.
You will inherit that `__init__`, but not the `__init__` generation.